measures/QOIReport/measure.rb

```diff
@@ -0,0 +1,304 @@
+    end
+
+    # Top 10 seasonal timing of peak (2)
+    seasons.each do |season, temperature_range|
+      next if season == Constants.SeasonOverlap
+
+      report_sim_output(runner, "average_of_top_ten_highest_peaks_timing_#{season}_hour", average_of_top_daily_timing(timeseries, temperature_range, "max"), "", "")
+    end
+
+    sqlFile.close
+
+    return true
+  end
+
+  def average_daily_use(timeseries, temperature_range, min_or_max)
+    daily_vals = []
+    timeseries["total_site_electricity_kw"].each_slice(24).with_index do |kws, i|
+      temps = timeseries["Temperature"][(24 * i)...(24 * i + 24)]
+      avg_temp = temps.inject { |sum, el| sum + el }.to_f / temps.size
+      if avg_temp > temperature_range[0] and avg_temp < temperature_range[1] # day is in this season
+        if min_or_max == "min"
+          daily_vals << kws.min
+        elsif min_or_max == "max"
+          daily_vals << kws.max
+        end
+      end
+    end
+    return daily_vals.inject { |sum, el| sum + el }.to_f / daily_vals.size
+  end
+
+  def average_daily_timing(timeseries, temperature_range, min_or_max)
+    daily_vals = []
+    timeseries["total_site_electricity_kw"].each_slice(24).with_index do |kws, i|
+      temps = timeseries["Temperature"][(24 * i)...(24 * i + 24)]
+      avg_temp = temps.inject { |sum, el| sum + el }.to_f / temps.size
+      if avg_temp > temperature_range[0] and avg_temp < temperature_range[1] # day is in this season
+        if min_or_max == "min"
+          daily_vals << kws.index(kws.min)
+        elsif min_or_max == "max"
+          daily_vals << kws.index(kws.max)
+        end
+      end
+    end
+    return daily_vals.inject { |sum, el| sum + el }.to_f / daily_vals.size
+  end
+
+  def average_of_top_daily_use(timeseries, temperature_range, min_or_max)
+    daily_vals = []
+    timeseries["total_site_electricity_kw"].each_slice(24).with_index do |kws, i|
+      temps = timeseries["Temperature"][(24 * i)...(24 * i + 24)]
+      avg_temp = temps.inject { |sum, el| sum + el }.to_f / temps.size
+      if avg_temp > temperature_range[0] and avg_temp < temperature_range[1] # day is in this season
+        if min_or_max == "min"
+          daily_vals << kws.min
+        elsif min_or_max == "max"
+          daily_vals << kws.max
+        end
+      end
+    end
+    daily_vals = daily_vals.sort.reverse
+    daily_vals = daily_vals[0...10]
+    return daily_vals.inject { |sum, el| sum + el }.to_f / daily_vals.size
+  end
+
+  def average_of_top_daily_timing(timeseries, temperature_range, min_or_max)
+    daily_vals = { "hour" => [], "use" => [] }
+    timeseries["total_site_electricity_kw"].each_slice(24).with_index do |kws, i|
+      temps = timeseries["Temperature"][(24 * i)...(24 * i + 24)]
+      avg_temp = temps.inject { |sum, el| sum + el }.to_f / temps.size
+      if avg_temp > temperature_range[0] and avg_temp < temperature_range[1] # day is in this season
+        if min_or_max == "min"
+          hour = kws.index(kws.min)
```

Comment:
To clarify, the way it's tracking hours is from 0 to 23 (because Ruby zero indexes, right)? So if I if "9" is the timing of a daily peak, it corresponds to 9 am?

Reply:
That's correct.